### PR TITLE
Use Kittyhawk for Deployment Config

### PR DESCRIPTION
**Pull request by joyliu-q**
Current blocked by https://github.com/pennlabs/infrastructure/pull/155
- [x] After PR above is merged, make sure to update kittyhawk version in package.json

This PR does the following:
- Transition from icarus to kittyhawk
- Update kraken and regenerate deployment CI
- Switch backend from `DOMAIN` to `DOMAINS` (list of domains supported)

To generate yaml, go into the `k8s` folder and run:
- `yarn build`

Checklist
- [x] package.json in `k8s` folder is uniform
- [x] package.json in `.github/cdk` folder is uniform 
- [x] renamed any mentions to `DOMAIN` in backend to `DOMAINS`
- [x] ran `yarn build` in `k8s` folder and compare new generated manifests w current

### CODE REVIEW

File: k8s/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "main": "main.js",
+  "types": "main.ts",
+  "license": "Apache-2.0",
+  "private": true,
+  "scripts": {
+    "import": "cdk8s import",
+    "synth": "cdk8s synth",
+    "compile": "tsc",
+    "watch": "tsc -w",
+    "test": "jest",
```
Comment (jonathankao97) [inline k8s/package.json:13]: also minor but not sure how this command is being used, not seeing any tests or jest config

**Review by jonathankao97 — APPROVED** [review]
minor changes but lgtm otherwise

File: k8s/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "build": "npm run compile && npm run synth",
+    "upgrade": "npm i cdk8s@latest cdk8s-cli@latest",
+    "upgrade:next": "npm i cdk8s@next cdk8s-cli@next"
```
[the remark this answers is not in the document]
Comment (joyliu-q) [reply k8s/package.json:16]: Running `yarn build` runs `npm` under the hood.

File: k8s/package.json (hunk 1 above)
Comment (joyliu-q) [reply k8s/package.json:13]: Good catch, jest was added automatically when creating the project but deleted later. Thanks!

**Pull request merged**